docs/components/table-view.md

```diff
@@ -0,0 +1,1039 @@
+/>
+
+```html
+<KTableView
+  :row-attrs="(row) => { return { 'data-testid': `row-${row.id}` } }"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### rowLink
+
+Function for turning row into a link. The function receives row value object as an argument and should return an object with two optional parameters:
+
+```ts
+interface RowLink {
+  to?: RouteLocationRaw | string // RouteLocationRaw or url string for row link
+  target?: '_self' | '_blank' | '_parent' | '_top' // Target for row link
+}
+```
+
+<KTableView
+  :row-link="getRowLink"
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```vue
+<template>
+  <KTableView
+    :row-link="getRowLink"
+    :data="tableData"
+    :headers="headers"
+  />
+</template>
+
+
+<script setup lang="ts">
+import type { RowLink } from '@kong/kongponents'
+
+const headers = [...]
+
+const tableData = [...]
+
+const getRowLink = (row: Record<string, any>): RowLink => ({
+  // using static route for demonstration purposes
+  // but you can generate dynamic routes based on the row data
+  to: { name: 'home' }, // pass a string to render the link as an anchor element instead of a router-link
+  target: '_blank',
+})
+</script>
+```
+
+### cellAttrs
+
+Function for adding custom attributes to each table cell. The function should return an object with `key: value` pairs for each attribute.
+
+The passed function receives an object with these parameters as an argument:
+```ts
+{
+  headerKey: string // header key
+  row: object // row value
+  rowIndex: number // row index
+  colIndex: index // column index
+}
+```
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders()"
+  :cell-attrs="({ headerKey, row }) => { return { 'data-testid': `column-${headerKey}-row-${row.id}` } }"
+/>
+
+```html
+<KTableView
+  :cell-attrs="({ headerKey, row }) => { return { 'data-testid': `column-${headerKey}-row-${row.id}` } }"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### maxHeight
+
+Limit the table height by passing a number, in pixels. If the table height exceeds the specified number, it will be scrollable. Table header is a `position: sticky;` element and will always be visible.
+
+<KTableView
+  max-height="300"
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  max-height="300"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### paginationAttributes
+
+Object to be passed to underlying pagination component. See [KPagination props](/components/pagination#props) for more details. Expects an object of type `TablePaginationAttributes`:
+
+```ts
+interface TablePaginationAttributes {
+  totalCount?: number
+  pageSizes?: number[]
+  initialPageSize?: number
+  currentPage?: number
+  offset?: boolean
+  disablePageJump?: boolean
+  offsetPreviousButtonDisabled?: boolean
+  offsetNextButtonDisabled?: boolean
+}
+```
+
+<KTableView
+  :pagination-attributes="{ totalCount: 100, currentPage: 3 }"
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
```

Comment:
issue: This example isn't functional; the current page never updates and changing the page size doesn't do anything

Reply:
Fixed

Reply:
The pagination isn't visible now?

Reply:
Sorry, made it visible & functional